### PR TITLE
Private events feature

### PR DESCRIPTION
**Pull request by Addis-Belete**
We asked the build a Private app. this app helps the user to create multiple events and also lets them attend many events.

### CODE REVIEW

**Review by ElijahTobs — CHANGES_REQUESTED** [review]
# Status: Changes Required :recycle: 

## Hi @Addis0943 and @luqmanmusah :wave: 

You've done well so far :+1:  and I commend you for that :clap: 
However, there are still some changes to be made before this milestone can be approved.

## REQUIRED

- [x] You should have a user's show page that displays events the user has created and events that the user is attending to separated in past and upcoming ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations)). There's currently no way to access the user's show page in your app.
- [x] You're expected to have navigation links are present at the top of the page ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/navigation_links_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
- [x] There should also be an attend button for past events

#### Controllers

##### EventsController

###### `index` action

* [x] lists ucomping events by using a class method on Event, i.e. `@past_events = Event.previous_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [x] lists ucomping events by using a class method on Event, i.e. `@upcoming_events = Event.upcoming_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

##### UsersController

###### `show` action

* [x] lists events created by user, i.e. `@created_events = current_user.events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/current_user_show_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [x] lists events previoulsy attendented by user **`using event model scope`**, i.e. `@past_events = current_user.attended_events.previous` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [x] lists upcoming events previously attended by user **`using event model scope`**, i.e. `@upcoming_events = current_user.attended_events.upcoming` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))




## Happy Coding :computer: 

_Feel free to reach out to me on slack_ **`@Elijah Ayandokun`**
  **Elijah O. Ayandokun**
[Github](https://github.com/ElijahTobs) | [LinkedIn](https://www.linkedin.com/in/ayandokunelijah/) | [Twitter](https://twitter.com/ElijahTobs)

**Comment by luqmanmusah** [comment]
Hello,
Fixed all the reviews stated above. Hoping to get approval soon.
Thanks

File: app/views/users/show.html.erb
Comment (acushlakoncept) [inline app/views/users/show.html.erb]: **[OPTIONAL]**
Note! this is considered optional because the previous reviewer may have missed it.
- [x] There should be absolutely NO logic in views. Views are only for displaying information that became available to it from its corresponding controller not generating it.  ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright)

**Don't know where to start or how to?** See  [Rails View Helpers Best Practices](https://medium.com/@acushla4real/rails-view-helpers-best-practices-guide-d7d7b6db3db4)

File: app/views/events/show.html.erb
Comment (acushlakoncept) [inline app/views/events/show.html.erb]: **[OPTIONAL]**
There should be absolutely NO logic in views. Move it to a helper file.
**Don't know where to start or how to?** See  [Rails View Helpers Best Practices](https://medium.com/@acushla4real/rails-view-helpers-best-practices-guide-d7d7b6db3db4)

**Review by acushlakoncept — CHANGES_REQUESTED** [review]
### Hi @Addis0943 and @luqmanmusah 👋

### You've done well so far 👍 and I commend you for that 👏

- [x] **This is just a suggestion:** When writing a project description, it's better to go ahead a describe what the project is about instead of starting it with "We were asked to build a Private app." (You had a typo) if you still feel like making a reference, at the end you could add "..... as part one of projects done at Microverse" or something.

- [x] I should not be able to create an empty event, kindly add validations.

#### I know the expectations of getting your project approved, but just add tests to your project as it will help you going forward.  It will be considered OPTIONAL since the previous reviewer may have missed it.

#### Tests

- [ ] models should have unit tests for validations and associations (Rspec) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/unit_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
- [ ] authentication should have integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
- [ ] event managment features should have integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))

### Cheers and Happy coding! 💻 👏

(Feel free to contact me on slack(@acushla) or reply to this thread, if you have any additional question)
Uduak Essien
[Twitter](https://twitter.com/acushlakoncept) | [LinkedIn](https://www.linkedin.com/in/acushlakoncept/)

**Comment by Addis-Belete** [comment]
Hello,
Fixed all the reviews stated above. Hoping to get approval soon.
Thanks

**Review by acushlakoncept — APPROVED** [review]
# Status: PROJECT APPROVED  🎉 🚀

### Hi @Addis0943 and @luqmanmusah  congratulations on completing your project. 👏 👏 👏 

I will go ahead and mark this as complete, feel free to merge when you are ready. All the best with the next project.

### Cheers and Happy coding! 💻 👏

(Feel free to contact me on slack(@acushla), if you have any additional question)
Uduak Essien
[Twitter](https://twitter.com/acushlakoncept) | [LinkedIn](https://www.linkedin.com/in/acushlakoncept/)

**Pull request merged**
